test/Microsoft.OpenApi.Tests/Validations/Visitors/OpenApiVisitorSetTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.OpenApi.Exceptions;
+using Microsoft.OpenApi.Models;
+using Xunit;
+
+namespace Microsoft.OpenApi.Validations.Visitors.Tests
+{
+    public class OpenApiVisitorSetTests
+    {
+        [Fact]
+        public void VisitorsPropertyReturnsTheCorrectVisitorList()
+        {
+            for (int i = 0; i < 5; i++) // 5 is just a magic number
+            {
+                // Arrange & Act
+                var visitors = OpenApiVisitorSet.Visitors;
+
+                // Assert
+                Assert.NotNull(visitors);
+                Assert.NotEmpty(visitors);
+                Assert.Equal(29, visitors.Count);
+            }
+        }
+
+        [Fact]
+        public void GetVisitorThrowsUnknowElementType()
+        {
+            // Arrange & Act
+            Action test = () => OpenApiVisitorSet.GetVisitor(typeof(OpenApiVisitorSetTests));
+
+            // Assert
+            var exception = Assert.Throws<OpenApiException>(test);
+            Assert.Equal("Can not find visitor type registered for type 'Microsoft.OpenApi.Validations.Visitors.Tests.OpenApiVisitorSetTests'.",
```

Comment:
>Can not find visitor type registered for type [](start = 26, length = 45)

This should be a public const in the main project and referenced here.

Reply:
@PerthCharern 

Yes. I can use the
`
Assert.Equal(`String.Format(SRResource.xxxxx, typeof(xxx).FullName)`, exception.Message);
`
However, it looks not good than the plain text in the test project. Because it can help us verify the error message thrown as our expected.  What do you think?

Reply:
@PerthCharern 

Yes. I can use the

Assert.Equal(String.Format(SRResource.xxxxx, typeof(xxx).FullName), exception.Message);

However, it looks not good than the plain text in the test project. Because it can help us verify the error message thrown as our expected.  What do you think?

---
In reply to: [157649686](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157649686) [](ancestors = 157649686)

Reply:
I think we should go with that instead of duplicating string literals. We are essentially checking that the exception thrown has the "right string", not that it is spelled correctly, etc.

I would say we get better verification from referring to the constant string in SRResource.

---
In reply to: [157832135](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157832135) [](ancestors = 157832135)

Reply:
Changed.

---
In reply to: [157882006](https://github.com/Microsoft/OpenAPI.NET/pull/152#discussion_r157882006) [](ancestors = 157882006,157832135)